source/extensions/health_checkers/http/health_checker_impl.cc

```diff
@@ -75,6 +75,30 @@ HttpHealthCheckerImpl::HttpHealthCheckerImpl(
       random_generator_(context.api().randomGenerator()) {
   // TODO(boteng): introduce additional validation for the authority and path headers
   // based on the default UHV when it is available.
+
+  // Process send payload.
+  if (config.http_health_check().has_send()) {
+    // Validate that the method supports a request body when payload is specified.
+    if (method_ == envoy::config::core::v3::GET || method_ == envoy::config::core::v3::HEAD) {
```

Comment:
TRACE also MUST NOT include a body ([RFC 7231](https://datatracker.ietf.org/doc/html/rfc7231#section-4.3.8)) and probably CONNECT,  all methods can be found [here](https://github.com/envoyproxy/envoy/blob/69b26409e012865fd957799470aa6f5a4d068d95/api/envoy/config/core/v3/base.proto#L44-L55).

Reply:
Added `TRACE` and `DELETE` to this list. `CONNECT` should be blocked at the config level.